Gordon360/Controllers/RecIM/TeamsController.cs

```diff
@@ -186,5 +186,32 @@ public ActionResult<IEnumerable<TeamInviteViewModel>> GetTeamInvites()
             var username = AuthUtils.GetUsername(User);
             return Ok(_teamService.GetTeamInvites(username));
         }
+
+        [HttpPatch]
+        [Route("{teamID}/accepted")]
```

Comment:
Similar to your work for 360 News, this route should reference a specific invite id, right? If you're not actually doing a series of changes to an invite object, I would refrain from passing that object from the frontend. Instead, it might be better to look it up based on an invite id.
```suggestion
        [Route("invites/{inviteID}/status")]
```
It doesn't need to follow this structure exactly, I'm not sure what the "accepted" status property is even called, but it is a reminder to focus routes around taking **actions** on **resources** in a rest API. An easy way to think about this case is: whenever you're making a PATCH or a PUT, you should end the URL in a specific ID of an object or an id then a property name like status.

Reply:
After I discussed with other teammates, we decided to stick with `teamID`. But we will use `teamID` and `ParticipantUsername` as PKs to look it up in the database.

Reply:
I understand the thought process, but this is not actually in line with the idea of a REST API. If you are using a composite key, you should really be referencing both pieces of the key in the URL, not implying them based on the current logged in user. If you want to use teamID and ParticipantUsername, that's totally fine, but I would then set the URL as follows:
```
[Route("{teamID}/invites/{username}/status")]
```
You may not have the idea in mind at any point to allow users to accept invites on behalf of another user, but REST has conventions for how to refer to its resources. They should be specified as a unique route to the resource, not based on authentication.

Reply:
Additionally, the `/invites` route really should have a username as a URL path or query string variable to filter rather than pulling invites for the current user from authentication. What if someday we decide that admins should be able to see invites for another user? It would be nice to not have to make potentially breaking changes to allow that.

Reply:
I'm not sure what your source is for saying that REST doesn't allow to take action on a resource based on authentication information, but I think it's a moot point here.

It's already an established convention in this codebase that the default action for resources pertains to the currently authenticated user, when not specified otherwise. See, for example, [`ProfilesController.Get()`](https://github.com/gordon-cs/gordon-360-api/blob/3afaa906ab191cd4954f19d506e680dfc6202c67/Gordon360/Controllers/ProfilesController.cs#L42).

Furthermore, I would argue this is a sensible and useful default, especially when there is no usecase for specifying an action on a different user.

> What if someday we decide

You could make the case that we might someday decide we need almost infinite features, but there's no point in trying to build for them now, as long as we aren't restricting ourselves to an architecture that will hold us back in the long run. We can trivially support that in the future with this current architecture, so there's no reason to add unneeded complexity for a feature that doesn't exist.

Reply:
> I'm not sure what your source is for saying that REST doesn't allow to take action on a resource based on authentication information, but I think it's a moot point here.

@EJPlatzer One of the main features of a rest API is having at least one unique URI to reference a resource. See [this article](https://www.restapitutorial.com/lessons/restfulresourcenaming.html), but I don't think you're confused on that.

Just because we have done something before does not mean it's the way we want to keep doing that thing. And I would refrain from calling something a convention unless it has been our intended practice in a bunch of places. Profiles is a very specific case and one that technically doesn't follow the REST unique naming architecture either.

> You could make the case that we might someday...
I totally agree that we don't want to jump through hoops to make future ideas possible, but there are many practices that developers use to make code more flexible for future changes. In this case, the update also aligns with REST API architecture so it seems like an easy win for code quality.

Reply:
@bennettforkner yes, I agree that providing a unique URI to reference the important resources is part of REST API design. That's not what I'm arguing with. Your original argument was:

>If you are using a composite key, you should really be referencing both pieces of the key in the URL, not implying them based on the current logged in user.

which does not follow from the idea that each resource should have a unique URI reference. It is true that our significant resources should have unique URIs, but that doesn't mean that we need to expose all possible data at all possible URIs - that would be extraordinarily cumbersome for both the API and it's consumer.

> Just because we have done something before does not mean it's the way we want to keep doing that thing. 

This is true, which is why I also argue that "this is a sensible and useful default, especially when there is no usecase for specifying an action on a different user."

> refrain from calling something a convention unless it has been our intended practice in a bunch of places.

We do, in fact, intentionally do this in a bunch of places throughout the API:
- Every route in the AcademicCheckInController
- DiningController
- EventsController.GetEventsByTerm
- HousingController: GetApplicationID, EditApplicaiton, ChangeEditor
- Every route in JobsController
- NewsController: GetNewsPersonalUnapprovedAsync, Post
- RequestsController.GetMembershipRequestsForCurrentUser
- StudentEmploymentController
- VictoryPromiseController
- WellnessController
and probably several of the unmaintained controllers relating to schedules and the messaging application.

> technically doesn't follow the REST unique naming architecture either

This isn't true for any useful perspective of REST. We are specifying the unique operation on these resources. For example, `GET /api/wellness` is inarguably a unique resource, it's just that action for that resource involves the currently authenticated user. There is no usecase for acting on that resource in relation to any other user, so it would actually be misleading to add a URI component for specifying the user.

> more flexible for future changes

I wasn't clear about this in my initial response, but the main reason I disagree that this is about flexibility is because there wouldn't be any breaking change involved in adding the functionality you describe, IF it ever became a feature. We simply create the new route that specifies the affected user, leaving untouched the original route that takes the default action relating to the authenticated user.

Your suggestion would result in a route that seems to allow changing the status of an invite for any arbitrary user, when in reality we only allow it for the authenticated user. This results in an objectively less consumer-friendly API.

Reply:
We're discussing offline btw

Reply:
@Kevin-Chu58 I think @EjPlatzer and I are in agreement now that we don't need to keep with REST API architecture in this case because of the convenience of referring to the invite as an object of the currently logged-in user. In that case, we would recommend a route that looks something like:
```
[Route("{teamID}/invite")]
```